src/coefficients/MFEMVectorFunctionCoefficient.C

```diff
@@ -11,18 +11,17 @@ MFEMVectorFunctionCoefficient::validParams()
 }
 
 MFEMVectorFunctionCoefficient::MFEMVectorFunctionCoefficient(const InputParameters & parameters)
-  : MFEMVectorCoefficient(parameters),
-    _func(getFunction("function")),
-    _vector_coefficient(3,
-                        [&](const mfem::Vector & p, double t, mfem::Vector & u)
-                        {
-                          libMesh::RealVectorValue vector_value =
-                              _func.vectorValue(t, PointFromMFEMVector(p));
-                          u[0] = vector_value(0);
-                          u[1] = vector_value(1);
-                          u[2] = vector_value(2);
-                        })
+  : MFEMVectorCoefficient(parameters), _func(getFunction("function"))
 {
+  _vector_coefficient = std::make_shared<mfem::VectorFunctionCoefficient>(
```

Comment:
Prefer member initialiser for this - [C.49](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c49-prefer-initialization-to-assignment-in-constructors)